google-cloud-firestore/lib/google/cloud/firestore/service.rb

```diff
@@ -172,6 +180,17 @@ def inspect
           "#{self.class}(#{@project})"
         end
 
+        def read_time_to_timestamp read_time
+          return nil if read_time.nil?
+
+          raise TypeError, "read_time is expected to be a Time object" unless read_time.is_a? Time
```

Comment:
Thinking about this some more, I'm now actually inclined not even to have this current check here, but just let it say `NoMethodError: usec` if users pass a non-Time-compatible object. Ruby types are duck-types, and it's not common to check for specific classes. But we can just leave it in for now; removing it later is not a breaking change.